### PR TITLE
PHPUnit update

### PR DESCRIPTION
**Pull request by kokspflanze**
- PHPUnit_Framework_TestCase migration to namespaces
- drop php5.5
- update phpunit version
- setExpectedException updated to expectException
- changed getMock to getMockBuilder
- fix assertions errors
- removed ServerTest::testSetObjectThrowsExceptionWithBadInput3, no tests there
- travis updated with lowest/locked and latest test
- drop hhvm/php 5.5 from build matrix and added php 7.1

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -13,15 +13,15 @@
         }
     },
     "require": {
-        "php": "^5.5 || ^7.0",
+        "php": "^7.0 || ^5.6",
```
Comment (froschdesign) [inline composer.json]: Please use: `"^5.6 || ^7.0"` (`"lowest || highest"`)

Reference:

* https://getcomposer.org/doc/articles/versions.md#tags
* https://github.com/zendframework/maintainers/pull/15#issuecomment-315531421

File: composer.json [hunk 2]
```diff
@@ -13,15 +13,15 @@
...
         "zendframework/zend-server": "^2.6.1",
         "zendframework/zend-stdlib": "^2.7 || ^3.0",
         "zendframework/zend-uri": "^2.5.2"
     },
     "require-dev": {
         "zendframework/zend-config": "^2.6",
         "zendframework/zend-http": "^2.5.4",
-        "phpunit/PHPUnit": "^4.8",
+        "phpunit/PHPUnit": "^6.2.1 || ^5.7.15",
```
Comment (froschdesign) [inline composer.json]: Same here.

**Comment by froschdesign** [comment]
A template for `.travis.yml` can be found here: https://github.com/zendframework/maintainers/blob/master/template/.travis.yml

File: composer.json (hunk 1 above)
Comment (kokspflanze) [inline composer.json]: nice to know, last time i had to change it to `highest || lowest`

**Comment by kokspflanze** [comment]
@froschdesign `travis.yml` should be the same now, expected the php5.6 and 7 tests

File: composer.json [hunk 3]
```diff
@@ -13,15 +13,15 @@
...
+        "php": "^5.6 || ^7.0",
         "zendframework/zend-server": "^2.6.1",
         "zendframework/zend-stdlib": "^2.7 || ^3.0",
         "zendframework/zend-uri": "^2.5.2"
     },
     "require-dev": {
         "zendframework/zend-config": "^2.6",
         "zendframework/zend-http": "^2.5.4",
-        "phpunit/PHPUnit": "^4.8",
+        "phpunit/PHPUnit": "^5.7.15 || ^6.2.1",
```
Comment (webimpress) [inline composer.json]: You can use here the latest version: ^5.7.21 || ^6.3

File: .travis.yml [hunk 4]
```diff
@@ -2,61 +2,75 @@ sudo: false
...
 
 env:
   global:
-    - SITE_URL: https://zendframework.github.io/zend-soap
-    - GH_USER_NAME: "Matthew Weier O'Phinney"
-    - GH_USER_EMAIL: matthew@weierophinney.net
-    - GH_REF: github.com/zendframework/zend-soap.git
-    - secure: "O2ECNTc7cif3sN3Iyjiheu4UsHcs4g/TrNtc5XqcTpecxcM4ncG1oLAoF62ZN2cwdpJ8nBwl3BQDI9nkLS+nxPHxdqV5bobZ/8ZcI1/2by+l9N720WQfezPxxSxAAPl5b1SXxocsWUiykJJpADqGdYjgfmgAyiK70tXN4XwZl27Svnh+ulqFB2aYQNUv28jJZ3/Pt8jW5sgTVYKYZ65bYirEHGMsUnOK5nJ/W04qoq1d8Qj5x4xTrA71KecibljXpqXHtrQRBpGoKmU8N0w7XdGn0l6xojEJ3QkLE1/1SrZq0r4GHgP294358IsepdQXoqAIxULR0nlNkLmDY//wQI6WRWOpTI+eYN+unoq/zDgPnZGzV6NX/BrRt1yxhNij8CpIvR8cg4WP52Cl2djG5YWr5sOOqffIvSDKJLZgUOb512F7DdI2Pr6HSDyOSqrKoMGMpLJb7/1sEumLfVGmpu9fgpWakxR5wTa0oMBZknyJVy18nj+mj5YgIzD+1FYzYj5ttajh3KKaxDgTCYHJkCA/MRONsr/Ap1oHI2W5iDgAEyaJhru0XefNjW37n/sx3QqZ+F4VZDbuTy8FaFMbnT6+brbCLdrxtmIwo2OhYob+G5rWQejnF81g1ShUE/H3l/rlQ+BLZzwYEgAHkJ5YTAQYq37E3Ra7DgOv14Ayjvo="
+    - COMPOSER_ARGS="--no-interaction"
+    - COVERAGE_DEPS="satooshi/php-coveralls"
```
Comment (webimpress) [inline .travis.yml]: We need here also legacy dependencies. Library still support PHP 5.6 and PHP 7 so it is possible to have PHPUnit 5 or 6 depends on PHP version.

File: composer.json (hunk 3 above)
Comment (kokspflanze) [inline composer.json]: the pr is some months ago, there was this uptodate

Comment (webimpress) [inline composer.json]: oh, yeh, you've just changed yday to `lowest || highest`, sorry.

File: .travis.yml (hunk 4 above)
Comment (kokspflanze) [inline .travis.yml]: when the older prs are okay, than i change all other prs. atm its bit strange to create a pr and 2 or 3 or 4 months later something changed and i have to update all once. this take a bit time.


also that lock file is generate with php5.6 so its not needed, have to change everything but i want a confirmation of the old prs like https://github.com/zendframework/zend-stdlib/pull/74

**Comment by kokspflanze** [comment]
travis error `The command "phpenv global 7.1" failed and exited with 1 during .` =(

**Pull request closed without merging**